From 1ed69e46e27b9085507e5498dbcc354dd5563338 Mon Sep 17 00:00:00 2001
From: Petr Oros <poros@redhat.com>
Date: Tue, 12 Sep 2017 16:29:48 +0200
Subject: [PATCH] cpuinfo: Fix compiler warnings in cpuinfo code

Signed-off-by: Petr Oros <poros@redhat.com>
---
 src/core/cpuinfo.cc | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/core/cpuinfo.cc b/src/core/cpuinfo.cc
index c617ef2..0fb0ea3 100644
--- a/src/core/cpuinfo.cc
+++ b/src/core/cpuinfo.cc
@@ -136,7 +136,7 @@ string value)
       cpu->claim(true);
       cpu->setVendor(s390x_vendor);
 
-      for(int i=0; i < s390x_features.size(); i++)
+      for(size_t i=0; i < s390x_features.size(); i++)
         cpu->addCapability(s390x_features[i]);
       /* many thanks to Martin Schwidefsky for communicating the descriptions
          of the feature flags
@@ -258,7 +258,7 @@ static void cpuinfo_aarch64(hwNode & node,
               if (node.getDescription() == "")
                 node.setDescription(aarch64_processor_name);
               cpu->claim(true);
-              for(int i=0; i < aarch64_features.size(); i++)
+              for(size_t i=0; i < aarch64_features.size(); i++)
                 {
                   cpu->addCapability(aarch64_features[i]);
                   cpu->describeCapability("fp", "Floating point instructions");
@@ -279,8 +279,6 @@ static void cpuinfo_ia64(hwNode & node,
 string id,
 string value)
 {
-  unsigned long long frequency = 0;
-  int i;
 
   if (id == "processor")
     currentcpu++;
-- 
2.14.1

