From 4af2308ad459924e792f4a09a7374ebae0cde805 Mon Sep 17 00:00:00 2001
From: Lyonel Vincent <lyonel@ezix.org>
Date: Sat, 23 Apr 2016 01:36:56 +0200
Subject: [PATCH 02/43] fix tar archive to avoid spilling .spec file outside of
 the source tree

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index b6d1297..8fe665e 100644
--- a/Makefile
+++ b/Makefile
@@ -16,5 +16,5 @@ $(PACKAGENAME).spec: $(PACKAGENAME).spec.in
 
 release: $(PACKAGENAME).spec
 	git archive --prefix=$(PACKAGENAME)-$(VERSION)/ -o $(PACKAGENAME)-$(VERSION).tar HEAD
-	tar -rf $(PACKAGENAME)-$(VERSION).tar $^
+	tar --transform s!^!$(PACKAGENAME)-$(VERSION)/! -rf $(PACKAGENAME)-$(VERSION).tar $^
 	gzip -f $(PACKAGENAME)-$(VERSION).tar
-- 
2.10.2

