From 72c611831f030ecabf5fbb34f1d0379b5897086c Mon Sep 17 00:00:00 2001
From: Lyonel Vincent <lyonel@ezix.org>
Date: Sat, 23 Apr 2016 10:03:23 +0200
Subject: [PATCH 03/43] update .gitignore

---
 .gitignore | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/.gitignore b/.gitignore
index e0292b1..cb79b25 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,2 +1,4 @@
 *.o
 *.a
+src/lshw
+lshw.spec
-- 
2.10.2

