From c818b7e2c4702d9b79bce644822183a40c4941ea Mon Sep 17 00:00:00 2001
From: Lyonel Vincent <lyonel@ezix.org>
Date: Fri, 19 Aug 2016 13:35:44 +0200
Subject: [PATCH 14/43] add detection of VirtualBox disks

---
 src/core/heuristics.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/core/heuristics.cc b/src/core/heuristics.cc
index 17d2dfa..45b39a4 100644
--- a/src/core/heuristics.cc
+++ b/src/core/heuristics.cc
@@ -109,6 +109,7 @@ static const char *disk_manufacturers[] =
   "FIREBALL.+", "Quantum",
   "^WDC.+", "Western Digital",
   "WD.+", "Western Digital",
+  "^VBOX.+", "VirtualBox",
   NULL, NULL
 };
 
-- 
2.10.2

